relay-server/src/services/processor.rs

```diff
@@ -1657,15 +1656,16 @@ impl EnvelopeProcessorService {
         };
 
         if let Some(outcome) = sampling_result.into_dropped_outcome() {
-            // Extract metrics here, we're about to drop the event/transaction.
-            self.extract_transaction_metrics(state, SamplingDecision::Drop, profile_id)?;
-
             let keep_profiles = dynamic_sampling::forward_unsampled_profiles(state, &global_config);
-
             // Process profiles before dropping the transaction, if necessary.
-            if keep_profiles {
-                profile::process(state, &self.inner.config);
-            }
+            // Before metric extraction to make sure the profile count is reflected correctly.
+            let profile_id = match keep_profiles {
```

Comment:
Is this easier to read than `if..else`?

Reply:
One line less!

Reply:
One line less and personally for me it reads nicer.